src/Utils.js

```diff
@@ -399,6 +399,17 @@ class Utils {
     }
     return obj;
   }
+
+  /**
+   * Encodes a string to be used in a URL.
+   * @param {String} input The string to encode.
+   * @returns {String} The encoded string.
+   */
+  static encode(input) {
+    return encodeURIComponent(input).replace(/[!'()*]/g, char =>
```

Comment:
From [here](https://github.com/parse-community/parse-server/issues/9111#issuecomment-2110890743):

> But GMail/GitHub/Facebook (etc) require escaping because people often put a closing parenthesis or square bracket or comma or period (etc) right after a URL in a sentence. In almost all cases, the URL didn't end with that punctuation, the punctuation was added by the user after pasting the URL. So the problem is that encodeURIComponent strictly follows the standard instead of dealing with certain non-conforming aspects of reality (that have good reasons to be non-conforming, imho).

How does it handle dot, comma, etc after the URL? Do we need to add these chars to the regex, or add tests for that?

Reply:
Expanded